### PR TITLE
Task-19: FE: Implement global toast component

### PR DESCRIPTION
**Pull request by denys-rybachok**

### CODE REVIEW

File: src/components/Toast/Toast.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  let background = '';
+  if (toast?.type === ToastEnum.ERROR) {
+    background = '#ba000d';
+  } else if (toast?.type === ToastEnum.SUCCESS) {
+    background = '#6A983C';
+  } else if (toast?.type === ToastEnum.WARNING) {
+    background = '#ed6c02';
+  } else {
+    background = '#0288d1';
+  }
```
Comment (VYaniv) [inline src/components/Toast/Toast.tsx]: Switch case and can be in the separate function outside the component.

File: src/components/btns/AuthBtn/AuthBtn.tsx [hunk 2]
```diff
@@ -28,9 +31,14 @@ const AuthBtn = () => {
...
-  const handleMenuItemClick = (menuItem: string): void => {
+  const handleMenuItemClick = async (menuItem: string): Promise<void> => {
     if (menuItem === UserMenuOptions.LOGOUT) {
-      dispatch(logoutUser());
+      try {
+        await dispatch(logoutUser()).unwrap();
+        dispatch(openToast({message: SuccessToastMessages.LOGGED_OUT, type: ToastEnum.SUCCESS}));
+      } catch (error) {
+        typeof error === 'string' && dispatch(openToast({message: error, type: ToastEnum.ERROR}));
+      }
```
Comment (VYaniv) [inline src/components/btns/AuthBtn/AuthBtn.tsx]: It should be separate method, and also switch/case will be better if talk about extending the app with other menu items.

File: src/components/forms/AuthForm/AuthForm.tsx [hunk 3]
```diff
@@ -17,20 +20,25 @@ const AuthForm = () => {
...
 
   const handleSignInFormSubmit = async (signInForm: SignInSchemaType): Promise<void> => {
-    const result = await dispatch(loginUser(signInForm)).unwrap();
-    if (!result.error) {
-      closeDialogAndGetUser();
+    try {
+      await dispatch(loginUser(signInForm)).unwrap();
+      closeDialogAndGetUser(SuccessToastMessages.LOGIN);
+    } catch (error) {
+      typeof error === 'string' && dispatch(openToast({message: error, type: ToastEnum.ERROR}));
```
Comment (VYaniv) [inline src/components/forms/AuthForm/AuthForm.tsx]: `typeof error === 'string'` looks strange, you can`t do such checking everywhere.

Comment (denys-rybachok) [reply src/components/forms/AuthForm/AuthForm.tsx]: Ok, thanks. It is for TS only, but I will try to do smth another.

File: src/components/Toast/Toast.tsx (hunk 1 above)
Comment (denys-rybachok) [reply src/components/Toast/Toast.tsx]:
```
const getBackgroundByToastType = (type: ToastEnum): string => {
  switch (type) {
    case ToastEnum.ERROR:
      return '#ba000d';
    case ToastEnum.SUCCESS:
      return '#6A983C';
    case ToastEnum.WARNING:
      return '#ed6c02';
    default:
      return '#0288d1';
  }
};
```

File: src/components/btns/AuthBtn/AuthBtn.tsx (hunk 2 above)
Comment (denys-rybachok) [reply src/components/btns/AuthBtn/AuthBtn.tsx]:
```
  const handleMenuItemClick = async (menuItem: UserMenuOptions): Promise<void> => {
    switch (menuItem) {
      case UserMenuOptions.LOGOUT:
        await handleLogout();
        break;
      case UserMenuOptions.PROFILE:
        handleOpenProfile();
        break;
      default:
        break;
    }
  };
```

File: src/components/forms/AuthForm/AuthForm.tsx (hunk 3 above)
Comment (denys-rybachok) [reply src/components/forms/AuthForm/AuthForm.tsx]: Changed to:
```
dispatch(openToast({message: error as string, type: ToastEnum.ERROR}));
```

Comment (VYaniv) [reply src/components/forms/AuthForm/AuthForm.tsx]: Ok, we will discuss it today. `error as string` or `error.toString()`

**Review by VYaniv — APPROVED** [review]

**Pull request merged**